src/main.test.js

```diff
@@ -0,0 +1,99 @@
+const { startCreating } = require('./main');
+const { isDnaUnique, createDna  } = require('./dna');
+const { needsExclusion } = require('./exclusions');
+
+jest.mock('fs', () => {
+  const originalModule = jest.requireActual('fs');
+
+  return {
+    ...originalModule,
+    mkdirSync: () => {},
+    writeFileSync: () => {},
+    readdirSync: () => (['a.png', 'b.png']),
+  };
+});
+
+jest.mock('canvas', () => {
+  const originalModule = jest.requireActual('canvas');
+
+  return {
+    ...originalModule,
+    createCanvas: jest.fn(() => ({
+      getContext: jest.fn(() => ({
+        clearRect: jest.fn(),
+        fillRect: jest.fn(),
+        drawImage: jest.fn(),
+      })),
+      toBuffer: jest.fn(),
+    })),
+    loadImage: jest.fn(() => ({})),
+  };
+});
+
+jest.mock('./config', () => {
```

Comment:
we mock the real config and some other functionality to simplify the test execution